skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,299 @@
+import numpy as np
+import scipy as sp
+
+
+class TPSTransform:
+    def __init__(self):
+        self._estimated =  False
+        self.parameters = np.array([], dtype=np.float32)
+        self.control_points = np.array([], dtype=np.float32)
+
+
+    def __call__(self, points):
+        """Map the source points to the destination surface.
+
+        Parameters
+        ----------
+        new_coords : array_like
+            Array of source points to be transformed
+        Returns
+        -------
+            ndarray : Mapped point of the distination point
+        """
+        x_src = _ensure_2d(points)
+
+
+        if x_src.shape[1] != self.control_points.shape[1]:
+            raise ValueError("Array must be identical")
+
+        dist = self._radial_distance(x_src)
+        transforms = np.hstack([dist, np.ones((x_src.shape[0], 1)), x_src])
+        return transforms @ self.parameters
```

Comment:
I think the implementation of this function still has a few kinks. I would expect this test to pass

```python
import skimage as ski
from_points = np.array([[0, 0], [0, 512], [512, 512],[512, 0]])
to_points = np.roll(from_points, 1, axis=0)

tform = ski.transform.TPSTransform()
tform.estimate(from_points, to_points)
result = tform(from_points)
np.testing.assert_allclose(result, to_points)
```
which raises
```
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0
Mismatched elements: 8 / 8 (100%)
Max absolute difference: 512.00000005
Max relative difference: 1.
 x: array([[ 3.541837e-10,  5.120000e+02],
       [ 5.120000e+02,  5.120000e+02],
       [ 5.120000e+02, -5.382495e-08],
       [ 6.430772e-13,  1.705303e-12]])
 y: array([[512,   0],
       [  0,   0],
       [  0, 512],
       [512, 512]])
```

Would also be useful to add this snippet as its own unit test.

Reply:
Also running 
```python
import skimage as ski
import numpy as np
import matplotlib.pyplot as plt
import scipy as sp

# Load image
image = ski.color.rgb2gray(ski.data.astronaut())
output_region = (0, 0) + image.shape

# Rotation by 90 degrees, counter-clockwise
from_points = np.array([[0, 0], [0, 512], [512, 512],[512, 0]])
to_points = np.roll(from_points, 1, axis=0)

tform = ski.transform.TPSTransform()
tform.estimate(from_points, to_points)
result = ski.transform.warp(image, tform)

expected = ski.transform.rotate(image, 90)

fig, ax = plt.subplots(ncols=3)
ax[0].imshow(image)
ax[0].set_title("original")
ax[1].imshow(result)
ax[1].set_title("result")
ax[2].imshow(expected)
ax[2].set_title("expected")
fig.show()
```
will return

![image](https://github.com/scikit-image/scikit-image/assets/20140352/d83a1ab2-ebf0-4583-b798-cf947f4656f2)

Reply:
But we should probably focus on the simpler example in https://github.com/scikit-image/scikit-image/pull/7040/files#r1271468801 for now.

Reply:
Upon updating the implementation to be consistent with other transform clase, this doesn't run.